### PR TITLE
[Core] convert gcs port read from env variable from str to int

### LINKED ISSUE #33963: [Core] os.getenv(ray_constants.GCS_PORT_ENVIRONMENT_VARIABLE)-- returned by getenv() is a string, but ray was not converted to int when used
### What happened + What you expected to happen

type error when ray read GCS_PORT_ENVIRONMENT_VARIABLE  from  os.environ

File c:\env39\lib\site-packages\ray\_private\client_mode_hook.py:105, in client_mode_hook.<locals>.wrapper(*args, **kwargs)
    103     if func.__name__ != "init" or is_client_mode_enabled_by_default:
    104         return getattr(ray, func.__name__)(*args, **kwargs)
--> 105 return func(*args, **kwargs)

File c:\env39\lib\site-packages\ray\_private\worker.py:1451, in init(address, num_cpus, num_gpus, resources, object_store_memory, local_mode, ignore_reinit_error, include_dashboard, dashboard_host, dashboard_port, job_config, configure_logging, logging_level, logging_format, log_to_driver, namespace, runtime_env, storage, **kwargs)
   1409     ray_params = ray._private.parameter.RayParams(
   1410         node_ip_address=node_ip_address,
   1411         raylet_ip_address=raylet_ip_address,
   (...)
   1445         node_name=_node_name,
   1446     )
   1447     # Start the Ray processes. We set shutdown_at_exit=False because we
   1448     # shutdown the node in the ray.shutdown call that happens in the atexit
   1449     # handler. We still spawn a reaper process in case the atexit handler
   1450     # isn't called.
-> 1451     _global_node = ray._private.node.Node(
   1452         head=True,
   1453         shutdown_at_exit=False,
   1454         spawn_reaper=True,
   1455         ray_params=ray_params,
   1456     )
   1457 else:
   1458     # In this case, we are connecting to an existing cluster.
   1459     if num_cpus is not None or num_gpus is not None:

File c:\env39\lib\site-packages\ray\_private\node.py:287, in Node.__init__(self, ray_params, head, shutdown_at_exit, spawn_reaper, connect_only, default_worker)
    285 # Start processes.
    286 if head:
--> 287     self.start_head_processes()
    289 if not connect_only:
    290     self.start_ray_processes()

File c:\env39\lib\site-packages\ray\_private\node.py:1119, in Node.start_head_processes(self)
   1116 if self._ray_params.external_addresses is not None:
   1117     self._redis_address = self._ray_params.external_addresses[0]
-> 1119 self.start_gcs_server()
   1120 assert self._gcs_client is not None
   1121 self._write_cluster_info_to_kv()

File c:\env39\lib\site-packages\ray\_private\node.py:924, in Node.start_gcs_server(self)
    922 """Start the gcs server."""
    923 gcs_server_port = self._ray_params.gcs_server_port
--> 924 assert gcs_server_port > 0
    925 assert self._gcs_address is None, "GCS server is already running."
    926 assert self._gcs_client is None, "GCS client is already connected."

TypeError: '>' not supported between instances of 'str' and 'int'

### Versions / Dependencies

In [53]: ray.__version__
Out[53]: '2.3.1'

\ray\_private\node.py, line 274 :
        if head:
            gcs_server_port = os.getenv(ray_constants.GCS_PORT_ENVIRONMENT_VARIABLE)
            if gcs_server_port:
                ray_params.update_if_absent(gcs_server_port=gcs_server_port)
            if ray_params.gcs_server_port is None or ray_params.gcs_server_port == 0:
                ray_params.gcs_server_port = self._get_cached_port("gcs_server_port")

### Reproduction script

import os,ray

os.environ["RAY_GCS_SERVER_PORT"]="32321" # !! os.environ.__setitem__ or putenv require the value to be of string type

ray.init()

### Issue Severity

None

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

convert the variable from str to int to close #33963

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
Duplicate: https://github.com/ray-project/ray/pull/34367

**Review by rickyyx — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/primevideo/">Amazon Prime Video</a> on <a href="https://giphy.com/gifs/primevideo-2020-borat-subsequent-moviefilm-Od0QRnzwRBYmDU3eEO">GIPHY</a></div>

File: python/ray/tests/test_basic_5.py [hunk 1]
```diff
@@ -361,6 +361,17 @@ def verify_imports(latch):
     ray.get(futures)
 
 
+def test_gcs_port_env():
+    try:
+        os.environ["RAY_GCS_SERVER_PORT"] = "12345"
+        ray.init()
+    except RuntimeError:
+        pass
```
Comment (cadedaniel) [inline python/ray/tests/test_basic_5.py:369]: nit: assert string value has some correct value so we don't hide other issues?

File: python/ray/tests/test_basic_5.py [hunk 2]
```diff
@@ -361,6 +361,17 @@ def verify_imports(latch):
     ray.get(futures)
 
 
+def test_gcs_port_env():
+    try:
+        os.environ["RAY_GCS_SERVER_PORT"] = "12345"
```
Comment (cadedaniel) [inline python/ray/tests/test_basic_5.py:366]: nit: use patch so state modification is limited to this test? example https://github.com/ray-project/ray/blob/0dafee965c96c51dd118bb56dab448cbfdeb2bb1/python/ray/tests/test_ray_debugger.py#L343

**Review by cadedaniel — APPROVED** [review]

**Pull request merged**